pytensor/tensor/extra_ops.py

```diff
@@ -2018,6 +2018,42 @@ def broadcast_with_others(a, others):
     return brodacasted_vars
 
 
+def concat_with_broadcast(tensor_list, dim=0):
+    """
+    Concatenate a list of tensors, broadcasting the non-concatenated dimensions to align.
+    """
+    dim = dim if dim > 0 else tensor_list[0].ndim + dim
```

Comment:
You should assert everything has the same ndim. Then use the numpy normalize_axis_index to handle the negative indices woth useful error message